src/OrchardCore.Modules/OrchardCore.Media/Assets/js/MediaApp/fileupload.js

```diff
@@ -12,6 +12,21 @@ $(document).on('mediaApp:ready', function () {
                 { name: '__RequestVerificationToken', value: antiForgeryToken },
             ]
         },
+        change: function (e, data) {
+            const maxSizeLimit = parseInt($('#maxSizeLimit').val());
+            var exceededFilesList = [];
+
+            $.each(data.files, function (index, file) {
+                if (file.size > maxSizeLimit) {
+                    exceededFilesList.push(index);
+                    mediaApp.errors.push(`The file '${file.name}' exceeds the limit.`);
```

Comment:
Does this string need to be localized? 
Did you check to see if the file upload plugin batches multiple files at the same time? 
I thought it does, and looks like it does here, so if it does the total size of all the files will be what the server errors on.

Think you need to add up the total size, and just error out if it exceeds what the server will take.

But you have tested it by lowering the max file size limit and tried it with some files right?

Reply:
> Does this string need to be localized?

Yes

>Did you check to see if the file upload plugin batches multiple files at the same time?
I thought it does, and looks like it does here, so if it does the total size of all the files will be what the server errors on.

Yes, but I'm not sure about the total batch, is there a max limit for a batch?

Reply:
> So I would have imagined just a variable at the top would be sufficent? and (probably) more in keeping with the Orchard Core Razor code.

I see

Reply:
> Yes, but I'm not sure about the total batch, is there a max limit for a batch?

That's why I suggested using the blueimp validation plugin that goes with the fileupload plugin this already uses.

Probably check their docs out about batch size, or just open your network tab, and see if it is batching file uploads, by uploading a few at once. If it is one upload then that's the size that's going to trigger the `MaxFileSize` value.

Reply:
OK